### ISSUE TITLE
Pushd do not cause error if switching to parent directory

### ISSUE THREAD

**Issue by rugk** [koalaman/shellcheck#1470]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2164, SC2164
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

pushd .. 
pushd ".."
pushd "hjh"
pushd .
```

#### Here's what shellcheck currently says:
```
Line 5	SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
Line 6	SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```
#### Here's what I wanted or expected to see:

```
Line 3	SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
Line 4	SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
Line 5	SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
Line 6	SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```

I.e. I see no reason why it should not also warn when I just switch to the parent directory. If I am in the root `/` this can likely fail, too…

Also it actually warns when switching to `.` (last line), so this would the only exception where I'd say, this can never fail.

**Comment by austin987** [comment]
Arguably `cd .` can fail; if the directory was removed by another process. Though it seems it still returns 0:
```
austin@laptop:/tmp/tmp.DQpNlBk45W/a$ cd . ; echo $?
cd: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
0

austin@laptop:/tmp/tmp.DQpNlBk45W/a/.$ pwd
/tmp/tmp.DQpNlBk45W/a/.
```

though the directory name changes then '/.' is appended.

**Comment by koalaman** [comment]
@austin987 I think that this may be related to the shell's path canonicalization more than your actual ability to `cd`.

`cd ..` in `/` is fine, it just leads you back to `/`. I think it'll only ever fail if the directory is removed or if you `chmod -x ..` or similar after going there.

I love the technical correctness of this suggestion. Pragmatically though, I think it's fair to assume that `cd ..` is sufficiently unlikely to fail that any warning is more noise than signal, and the warning for `cd ..` is explicitly and intentionally squashed.  `cd .` warning is a plain oversight, as is `cd ../..`. This has been fixed in e8f913e9. Additionally, 1d60a4a4 makes shellcheck say `pushd` instead of `cd` when that's the command you're using.

Thanks for the suggestion!

**Comment by rugk** [comment]
So `..` is intended and `.` has been fixed? Just because none of your commit hashes are linked - usually GitHub dos this automatically if it is a valid commit hash.

**Comment by koalaman** [comment]
My bad, I rebased due to a conflict without changing the hash. It's d4d219a and 489c3a4ddf, and the changes are already live on shellcheck.net
